static-assets/components/cstudio-forms/forms-engine.js

```diff
@@ -3615,11 +3613,8 @@ const initializeCStudioForms = () => {
               attributes.pop();
             }
 
-            // Because we added start and end elements, we can be sure that any field names will
-            // be delimited by the delimiter token (ie. comma)
-            fieldRe = new RegExp(',' + key + '(?:,||d+|)', 'g');
-
-            if (fieldRe.test(validFieldsStr)) {
+            // Check if the current form field key (id) is in the validFields array
+            if (validFields.includes(key)) {
```

Comment:
When checking for the valid field, we were validating with a regex but since we're checking against the form model, we can only check if the current field is in the validFields array.
There was an error in the regex causing that for example
`components` was valid because the regex watch matching a field like `components_o`, causing the field to end up incorrectly in the xml